js/api.js

```diff
@@ -1,8 +1,31 @@
+//@ts-check
 export const fetchCategories = async () =>
     fetch(`http://localhost:3000/categories`).then(response => response.json());
 
 export const fetchFeaturedProducts = (categoryId) =>
     fetch(`http://localhost:3000/items?categoryId=${categoryId}`).then(response => response.json());
 
 export const fetchCarouselProducts = () =>
-    fetch(`http://localhost:3000/items?flag=1`).then(response => response.json());
+    fetch(`http://localhost:3000/items?flag=on`).then(response => response.json());
+
+export const deleteProduct = (id) =>
+    fetch(`http://localhost:3000/items/${id}`, { method: 'DELETE' })
+
+export const createProduct = (body) =>
```

Comment:
I'd rename the `body` to `formData` -- it better reflects the underlying type that this method expects.